dashboard/src/components/api/che-workspace.factory.ts

```diff
@@ -475,7 +477,13 @@ export class CheWorkspace {
       this.websocketBusByWorkspaceId.set(workspaceId, bus);
 
       bus.subscribe('workspace:' + workspaceId, (message) => {
-        this.getWorkspaceById(workspaceId).status = message.eventType;
+
+        //Filter workspace events, which really indicate the status change:
+        if (this.workspaceStatuses.indexOf(message.eventType) >= 0) {
+          this.getWorkspaceById(workspaceId).status = message.eventType;
+        } else if (message.eventType === 'SNAPSHOT_CREATED') {
+          this.getWorkspaceById(workspaceId).status = 'RUNNING';
```

Comment:
yes, workspace snapshot can be created for running workspace only, so it's really running. But we artificially introduce - SNAPSHOT_CREATING - to display it as status, though real status is still RUNNING. And we get back to it on SNAPSHOT_CREATED event